src/meta/src/hummock/manager/mod.rs

```diff
@@ -1045,7 +1045,15 @@ where
             compact_task.compaction_group_id,
         );
 
-        self.try_send_compaction_request(compact_task.compaction_group_id)?;
+        if !CACEL_STATUS_SET.contains(&compact_task.task_status()) {
+            // We no need to reschedule when report cancel_task
+            // In fact, when `cancel_task` appears in the system, it means that the system
+            // processing capacity may have been overloaded, and we may no longer need to try to
+            // schedule. And this is a quick fix
+            // TODO: need to consider the fairness of the compaction_group and unload the schedule
+            // when overloaded
+            self.try_send_compaction_request(compact_task.compaction_group_id)?;
```

Comment:
I think it still makes sense to retry compaction when we have task cancelled in some cases:
- SendFailCanceled: this happens when the rpc between meta and a picked compactor fails but if there are other available compactors, we can still trigger compaction so skip this case brings no benefit.
- Same for HeartbeatCanceled

Given that the issue caused by rescheduling tasks when compactor is not available is partially fixed by #5405. I think we can skip this PR.

Reply:
I think with schedule state purposed in #5321, we can send the request regardless.